packages/state-transition/src/cache/epochCache.ts

```diff
@@ -788,15 +871,37 @@ export class EpochCache {
     return shuffling;
   }
 
+  getDecisionRoot(epoch: Epoch): RootHex {
+    switch (epoch) {
+      case this.epoch - 1:
+        return this.previousDecisionRoot;
+      case this.epoch:
+        return this.currentDecisionRoot;
+      case this.nextEpoch:
+        return this.nextDecisionRoot;
+      default:
+        throw new EpochCacheError({
+          code: EpochCacheErrorCode.DECISION_ROOT_EPOCH_OUT_OF_RANGE,
+          currentEpoch: this.epoch,
+          requestedEpoch: epoch,
+        });
+    }
+  }
+
   getShufflingAtEpochOrNull(epoch: Epoch): EpochShuffling | null {
-    if (epoch === this.previousShuffling.epoch) {
-      return this.previousShuffling;
-    } else if (epoch === this.currentShuffling.epoch) {
-      return this.currentShuffling;
-    } else if (epoch === this.nextShuffling.epoch) {
-      return this.nextShuffling;
-    } else {
-      return null;
+    switch (epoch) {
+      case this.epoch - 1:
+        return this.previousShuffling;
+      case this.epoch:
+        return this.currentShuffling;
+      case this.nextEpoch:
+        if (!this.nextShuffling) {
+          this.nextShuffling =
+            this.shufflingCache?.getSync(this.nextEpoch, this.getDecisionRoot(this.nextEpoch)) ?? null;
```

Comment:
can we get rid of this method in state-transition? beacon-node apis should use ShufflingCache instead

Reply:
which method? getShufflingAtEpochOrNull ?

Reply:
> can we get rid of this method in state-transition? beacon-node apis should use ShufflingCache instead

When I tracked the function stacks to look for side-effects of removal there were a couple places that I thought that nextShuffling might be pulled so I left it here to avoid the conditional to check the `EpoochCache` and the `ShufflingCache`.  I will try to remove it and update the API calls and then run the sim tests again to see what happens

> which method? getShufflingAtEpochOrNull ?

@wemeetagain its the pulling of `nextShuffling` from the `EpochCache` instead of directly from the `ShufflingCache`.  In theory the only time a shuffling should be pulled for the next epoch is to get duties ahead of time via the API.

Reply:
I could not remove the nextShuffling.  The `fork_choice/get_head/pyspec_tests > fork_choice/get_head/pyspec_tests/filtered_block_tree` spec test for all forks failed